### PR TITLE
Use etherscan host from job run metadata

### PR DESCRIPTION
**Pull request by rupurt**
[Fixes #166277095]

### CODE REVIEW

File: explorer/client/src/__tests__/reducers/config.test.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  })
+
+  it('can update the search query', () => {
+    const action = {
+      type: 'UPSERT_JOB_RUN',
+      data: {
+        meta: {
+          jobRun: {
+            meta: {
+              etherscanHost: 'ropsten.etherscan.io'
```
Comment (dimroc) [inline explorer/client/src/__tests__/reducers/config.test.ts]: Do we have to have so many `meta` nested in the payload?

```
data: { meta: { etherscanHost: ...
```

The above wouldn't work?

File: explorer/client/src/reducers/config.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  | { type: '@@redux/INIT' }
+  | { type: '@@INIT' }
+
+const initialState = { etherscanHost: undefined }
+
+export default (state: IState = initialState, action: Action) => {
+  switch (action.type) {
+    case 'UPSERT_JOB_RUN':
+      return Object.assign({}, state, {
+        etherscanHost: action.data.meta.jobRun.meta.etherscanHost
```
Comment (dimroc) [inline explorer/client/src/reducers/config.ts]: Two things:
1. Reiterate the long combination of keys: ` action.data.meta.jobRun.meta.etherscanHost`
2. Is `UPSERT_JOB_RUN` really the only action type we need to cover. If we aren't mutating, but merely retrieving, don't we still need the etherscanHost? Or is that all covered in this one action `UPSERT_JOB_RUN`? Seems so when I peruse https://github.com/smartcontractkit/chainlink/blob/master/explorer/client/src/reducers/jobRuns.ts#L25

File: explorer/client/src/__tests__/reducers/config.test.ts (hunk 1 above)
Comment (rupurt) [reply explorer/client/src/__tests__/reducers/config.test.ts]: I'm not a fan of the structure either. It's generated by [`normalize(...)`](https://github.com/yury-dymov/json-api-normalizer)

File: explorer/client/src/reducers/config.ts (hunk 2 above)
Comment (rupurt) [reply explorer/client/src/reducers/config.ts]: 1. `UPSERT_...` in this context refers to inserting or updating the job run in the redux store. After the store has the job run, it gets retrieved by a selector for display.
2. So far yes. If we wanted to show a link on the search results index for example, we would need to change the approach.

**Pull request merged**